annotationPipeline/src/main/java/org/cbioportal/annotation/pipeline/MutationFieldSetMapper.java

```diff
@@ -32,36 +32,67 @@
 
 package org.cbioportal.annotation.pipeline;
 
-import java.util.*;
 import org.cbioportal.models.MutationRecord;
+import java.util.*;
 import org.springframework.batch.item.file.mapping.FieldSetMapper;
 import org.springframework.batch.item.file.transform.FieldSet;
 import org.springframework.validation.BindException;
+import org.springframework.cglib.beans.BeanMap;
+import com.google.common.collect.Sets;
+import com.google.common.base.Strings;
+import java.util.stream.Collectors;
 
 /**
  *
  * @author heinsz
  */
 public class MutationFieldSetMapper implements  FieldSetMapper<MutationRecord> {
-
     @Override
     public MutationRecord mapFieldSet(FieldSet fs) throws BindException {
-        MutationRecord record = new MutationRecord();
-        Set<String> names = new HashSet(Arrays.asList(fs.getNames()));
-        names.addAll(record.getHeader());
+        MutationRecord mr = new MutationRecord();
+        BeanMap recordBeanMap = BeanMap.create(new MutationRecord());
+        Set<String> fieldSetNames = new HashSet(Arrays.asList(fs.getNames()));
+        Set<String> mutationRecordFields = new HashSet(mr.getHeader());
+        
+        // We need to intersect these two sets ignoring case, and then use this
+        // intersection to filter the original set for data accession
+        Set<String> intersection = Sets.intersection(
+            mutationRecordFields.stream()
+                .map(field -> field.toUpperCase())
+                .collect(Collectors.toSet()),
+            fieldSetNames.stream()
+                .map(field -> field.toUpperCase())
+                .collect(Collectors.toSet()));
+        
+        // Here we make sure that the field is in our intersection
+        Set<String> fields = fieldSetNames.stream()
+                .filter(line -> intersection.contains(line.toUpperCase()))
+                .collect(Collectors.toSet());
+        
+        // In this we actually set the value in our bean map
+        fields.stream().forEach(column -> 
+            {
+                recordBeanMap.put(column.toUpperCase(),Strings.isNullOrEmpty(fs.readRawString(column)) ? "" : fs.readRawString(column));
```

Comment:
Just something to keep in mind - reading the raw string will also include trailing white spaces. May want to trim just in case. Same [here](https://github.com/cBioPortal/genome-nexus-annotation-pipeline/pull/37/files?diff=split#diff-1178360edd668d8dd9cd51e477e1a623R94)